### PR TITLE
Add People API connector initial implementation

### PR DESCRIPTION
**Pull request by kapilraajP**
This pull request contains Basic implementation of Ballerina Google People API connector implementation.
Resolves wso2-enterprise/choreo#2764

### CODE REVIEW

File: endpoint.bal [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+        string readMask = prepareReadMaskString(readMasks);
+        json copyPayload = {
+            "copyMask": copyMask,
+            "readMask": readMask
+        };
+        request.setJsonPayload(copyPayload);
+        http:Response httpResponse = <http:Response>check self.googleContactClient->post(path, request);
+        var response = check handleResponse(httpResponse);
+        Person person = check response.cloneWithType(Person);
+        return person;
```
Comment (LakshanSS) [inline endpoint.bal:76]: Can't we do like this?
```
return check response.cloneWithType(Person);
```

File: tests/main_test.bal [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+        test:assertTrue(true, msg = "Batch Get Contact Group Failed");
+    } else {
+        test:assertFail(msg = batchGetContactGroup.message());
+    }
+}
+
+@test:Config {dependsOn: [testCreateContactGroup]}
+function testListContactGroup() {
+    log:print("Running List Contact Group Test");
+    runtime:sleep(10);
```
Comment (LakshanSS) [inline tests/main_test.bal:222]: Why are we doing this? Any reason?

File: utils.bal [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+
+# Prepare URL with for batch operations.
+# 
+# + resourceNames - String array of resource names
+# + return - The prepared URL string
+isolated function prepareResourceString(string pathReceived, string[] resourceNames) returns string {
+    string path= pathReceived;
+    int count = 0;
+    while (count < (resourceNames.length() - 1)) {
+        path = path + CONTACTGROUP_PATH + resourceNames[count] + "&";
```
Comment (LakshanSS) [inline utils.bal:143]: Can use constants for symbols like "&".
Check all the places for similar issues.

File: endpoint.bal (hunk 1 above)
Comment (kapilraajP) [reply endpoint.bal]: Can be done

File: endpoint.bal [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+    # Get connections
+    # 
+    # + personFields - Restrict which fields on the person are returned
+    # + options - Record that contains options
+    # + return - `stream<Person>` on success or else an `error`
+    remote function listPeopleConnection(string[] personFields, ContactListOptions? options = ()) returns 
+                                          @tainted stream<Person>|error {
+        string path = SLASH + LIST_PEOPLE_PATH;
+        string pathWithPersonFields = prepareUrlWithPersonFields(path, personFields);
+        io:println(pathWithPersonFields);
```
[the remark this answers is not in the document]
Comment (kapilraajP) [reply endpoint.bal]: removed

File: tests/main_test.bal (hunk 2 above)
Comment (kapilraajP) [reply tests/main_test.bal:222]: I removed from this place. Resource creation take small time. that is why added in other places

File: endpoint.bal [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+        string path = SEARCH_OTHERCONTACT_PATH + READ_MASK_PATH + QUERY_PATH + query;
+        http:Response httpResponse = <http:Response>check self.googleContactClient->get(path);
+        var response = check handleResponse(httpResponse);
+        return check response.cloneWithType(SearchResponse);
+    }
+
+    # Create a contact.
+    # 
+    # + createContact - Record of type of `CreatePerson`
+    # + personFields - Restrict which fields on the person are returnedy
```
Comment (LakshanSS) [inline endpoint.bal:91]: Typo 
"returned"

Comment (kapilraajP) [reply endpoint.bal:91]: i will correct

File: README.md [hunk 6]
```diff
@@ -1,2 +1,344 @@
...
+contacts:GoogleContactsConfiguration googleContactConfig = {
+    oauthClientConfig: {
+        clientId: clientId,
+        clientSecret: clientSecret,
+        refreshUrl: contacts:REFRESH_URL,
+        refreshToken: refreshToken
+    }
+};
+
+contacts:Client googleContactClient = checkpanic new (googleContactConfig);
```
[the remark this answers is not in the document]
Comment (kapilraajP) [reply README.md:90]: That make sense, I will use that

File: utils.bal [hunk 7]
```diff
@@ -0,0 +1,381 @@
...
+# Prepare URL with read masks.
+# 
+# + url - Url to be appended
+# + readMasks - String array of fields to be fetched
+# + return - The prepared URL string
+isolated function prepareUrlWithReadMasks(string url, string[] readMasks) returns string {
+    string path = url;
+    int count = 0;
+    while (count < (readMasks.length() - 1)) {
+        path = path + readMasks[count] + ",";
```
[the remark this answers is not in the document]
Comment (kapilraajP) [reply utils.bal:49]: Noted

File: constants.bal [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+const string SPACE = " ";
+const string SLASH = "/";
+const string COLON = ":";
+const string QUESTION_MARK = "?";
+const string AMBERSAND = "&";
+const string EQUAL = "=";
+# Constant for paths.
+const string CONTACT_GROUP_PATH = "contactGroups";
+const string BATCH_CONTACT_GROUP_PATH = ":batchGet?";
+const string LIST_PEOPLE_PATH = "people/me/connections?requestSyncToken=true&syncToken=MisAPB3nNAAAABIIvbPr36uh7wIQvbPr36uh7wJhJlAhhPtQDf5q8aaPtMVGOgw4NDkzNTA1ODUxNjY=&";
```
Comment (LakshanSS) [inline constants.bal:30]: Here a token is added in this string. Is this correct? 
Can u check about this?

File: tests/main_test.bal [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+        test:assertFail(msg = listContacts.message());
+    }
+}
+
+@test:Config {}
+function testCopyOtherContactToMyContact() {
+    log:print("Running copy OtherContact To MyContact Test");
+    string[] copyMask = ["names", "emailAddresses", "phoneNumbers"];
+    string[] readMask = ["names", "emailAddresses", "phoneNumbers"];
+    var copyContacts = googleContactClient->copyOtherContactToMyContact(copyMask, readMask, "otherContacts/c8846080985039646639");
```
Comment (LakshanSS) [inline tests/main_test.bal:52]: otherContacts/------
Is it a constant?
Check if this is correct.

File: tests/main_test.bal [hunk 10]
```diff
@@ -0,0 +1,281 @@
...
+    } else {
+        test:assertFail(msg = getContactGroup.message());
+    }
+}
+
+@test:Config {dependsOn: [testCreateContactGroup]}
+function testbatchGetContactGroup() {
+    log:print("Running BatchGet Contact Group Test");
+    runtime:sleep(10);
+    string[] resourceNames = [contactGroupResourceName, "contactGroups/a57e6fd8bd0c18f"];
```
Comment (LakshanSS) [inline tests/main_test.bal:211]: The value added aftrt contactGroups/
Is it a constant value?

File: constants.bal (hunk 8 above)
Comment (kapilraajP) [reply constants.bal:30]: It is no longer needed. removed

File: tests/main_test.bal (hunk 9 above)
Comment (kapilraajP) [reply tests/main_test.bal:52]: This thing need to be discussed

File: Module.md [hunk 11]
```diff
@@ -0,0 +1,8 @@
+
+## Compatibility
+
+|                               |      	      Version             |
+| :---------------------------: | :-----------------------------: |
+|      Ballerina Language       |        Swan Lake Alpha 2  	  |
+|      Google People API        |                V1               |
+
```
Comment (LakshanSS) [inline Module.md:8]: Now we need only package.md for central.
We can delete module.md

Comment (kapilraajP) [reply Module.md]: Deleted

**Comment by abeykoon** [comment]
Closing PR as implementation is deviated. We will directly do the initial implementation on Ballerina version `SLAlpha4`.

**Pull request closed without merging**